src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -223,6 +223,9 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_ReadyToRunOutputPath>$(IntermediateOutputPath)R2R</_ReadyToRunOutputPath>
     </PropertyGroup>
 
+    <!-- TODO: Temp error to limit R2R compilation to self-contained mode only. Delete this when fixing #3109 and #3110 -->
+    <NETSdkError Condition="$(SelfContained) == False" ResourceName="ReadyToRunNoValidRuntimePackageError" />
```

Comment:
I wanted to get your thoughts on the error message here... It's a bit confusing because it's not mentioning that the failure is because we're publishing a FX dependent app. This is however a very temporary situation, so I wasn't sure if it was worth adding another entry for it in Strings.resx. I'm happy to add a new entry if we think it's better, I just wanted to get your thoughts on it first.

Reply:
For reference, here's what the error is:

> NETSDK1095: Optimizing assemblies for performance is not supported for the selected target platform or architecture. Either set the ReadyToRun property to false, or use a supported runtime identifier when publishing.

Reply:
Can we just change the text of the signgle messages to enumerate all the current prerequisites, then amend it when we support fx-dependent? I think it should say something about self-contained currently being required.

Reply:
@nguerrera That's a good suggestion. We can amend the message and change it later